### ISSUE TITLE
[Question] Is this still maintained?

### ISSUE THREAD

**Issue by BananaHammock743** [wymangr/blueiris_exporter#4]
Will this project be maintained going forward?

**Comment by wymangr** [comment]
I'm happy to provide enhancements and bug fixes upon request, but I don't have anything planned out for this going forward. It's serving the purpose for me personally, and there hasn't been much interest.

**Comment by BananaHammock743** [comment]
Thanks for the reply. I love it (just reinstalled on my new BI rig). It's great to see a summary of BI instead of twarling through Logs. You should post it (again?) on r/blueiris with the corresponding Grafana dashboard, I'm sure you'd get a few more people using. 

One question: The AI section of the Grafana dashboard doesn't seem to be populating with CodeProject AI, should it be or is it only setup for DeepStack? Happy to make a new issue for CPAI, post logs, etc if needed

![image](https://github.com/wymangr/blueiris_exporter/assets/48776255/71a1b011-997f-4f32-9816-c9c88ceb5053)

Blue Iris: 5.7.6.8 x64 (19/05/23)
CPAI: 1.2.9 BETA

**Comment by wymangr** [comment]
I'm glad to hear you like it @BananaHammock743!  

It should work with both DeepStack and CodeProject AI (That's what I'm currently using). How did you setup the exporter? As a service or running from the command line? Are any other areas of the dashboard working? If you edit one of the panels, can you make sure that the data source is pointed to the correct Prometheus instance that is scraping the metrics? Can you view the metrics page in a web browser?

**Comment by BananaHammock743** [comment]
Details below. Yell out if you need any further detail

- I've tried two ways of running the exporter; 1, in the windows startup folder having a shortcut file to the .exe; 2. in the windows startup folder having a shortcut file to the .bat file, in the .bat file is just calling the .exe. Regardless of which option, the AI sections (mentioned below) don't populate. Is there a preferred way to have the exporter run?
- Sections which aren't populating: AI, AI Analysis, Blue Iris AI Duration Average (all other section is populated). 
- BI 5.7.6.8, CodeProject AI 2.1.9-Beta
- In Grafana? Yep I've checked each section heading and confirmed that is is pointing to the prometheus instance. 
- Yes, can view metrics page no problem

[2 days of logs.txt](https://github.com/wymangr/blueiris_exporter/files/11645566/2.days.of.logs.txt)
[metrics endpoint.txt](https://github.com/wymangr/blueiris_exporter/files/11645567/metrics.endpoint.txt)
![dashbaord screen1](https://github.com/wymangr/blueiris_exporter/assets/48776255/0b6e0be4-638a-423a-9438-04eee1e6f110)
![dashbaord screen2](https://github.com/wymangr/blueiris_exporter/assets/48776255/85cfedbc-aa56-48b4-a0f9-ac4cca9aea59)
![dashbaord screen3](https://github.com/wymangr/blueiris_exporter/assets/48776255/bc3f4ff4-9828-4d40-ab70-e684e839e698)
![dashbaord screen4](https://github.com/wymangr/blueiris_exporter/assets/48776255/571b9b6b-a7f2-495f-898d-ca433d114425)

**Comment by wymangr** [comment]
Thanks for the info @BananaHammock743 I'm looking into it and i'll get back to you when I figure it out :)

**Comment by wymangr** [comment]
@BananaHammock743 

Hmm.. so I was able to run it successfuly with the log file you provided without any issues. The AI panels in Grafana are created off of the `blueiris_ai_duration_distinct` metric  which will only show up on the /metrics page once per alert and will only show up again when a new alert comes in. I did confirm that that metric did populate and that they came back when I added a new alert at the bottom of your log file.

See attached metrics: 
[metrics.txt](https://github.com/wymangr/blueiris_exporter/files/11681518/metrics.txt)

For next troubleshooting steps, I would remove the executable from the startup fold and instead:
1. open a command prompt
2. change directory to where you saved the .exe file (`ex: cd c:\Users\USER\Downloads`)
3. run the file, making sure that if your log directory is not `C:\BlueIris\log\` that you specify the correct log path. (`blueiris_exporter-amd64.exe --logpath=C:\Some\Different\BlueIris\log`)

When you start it, don't go to the metrics page right away, let Prometheus do the first scrape. Once it does, check the Grafana dashboard. If it worked, you should see each camera's last analysis in the AI panels. If you still don't, check the command prompt window to see if there are any errors on the screen. If still nothing, try opening up the `Explore` page in Grafana and just query the `blueiris_ai_duration_distinct` metric. Maybe it's an issue with the Grafana dashboard?

**Comment by wymangr** [comment]
@BananaHammock743 Were you able to get it working?

**Comment by wymangr** [comment]
@BananaHammock743 Just wanted to follow up to see if you need anything else.

**Comment by wymangr** [comment]
@BananaHammock743 I think I found the issue. There is a problem with the dashboard I uploaded. 

There is an incorrect variable set to a version of the exporter that I had setup for testing. If you go to the dashboard, click on the dashboard settings, navigate to variables, and edit the "type" variable. Change the query to `label_values(blueiris_ai_duration, type)`

Edit: I added a new revision to the dashboard with the fix: https://grafana.com/api/dashboards/17432/revisions/4/download
